test/plugins/validation/2and3/items-required-for-array-objects.test.js

```diff
@@ -173,32 +173,46 @@ describe('validation plugin - semantic - items required for array objects - Open
     expect(res.warnings.length).toEqual(0);
   });
 
-  it('should return an error when a model does not define a required property', () => {
+  it('should return an warning when a model does not define a required property', () => {
     const spec = {
       components: {
         schemas: {
-          TestObject: {
-            type: 'object',
-            required: ['ImportantProperty'],
+          RefProp: {
             properties: {
               OptionalProperty: {
-                type: 'string'
+                type: `string`
               }
             }
+          },
+          TestObject: {
+            type: 'object',
+            required: ['ImportantProperty'],
+            oneOf: [
+              {
+                $ref: '#/components/schemas/RefProp'
+              },
+              {
+                properties: {
+                  ImportantProperty: {
+                    type: `string`
+                  }
+                }
+              }
+            ]
           }
         }
       }
     };
 
     const res = validate({ jsSpec: spec });
-    expect(res.errors.length).toEqual(1);
-    expect(res.errors[0].path).toEqual(
+    expect(res.errors.length).toEqual(0);
+    expect(res.warnings.length).toEqual(1);
+    expect(res.warnings[0].path).toEqual(
       'components.schemas.TestObject.required[0]'
     );
-    expect(res.errors[0].message).toEqual(
-      "Schema properties specified as 'required' must be defined"
+    expect(res.warnings[0].message).toEqual(
+      "Schema properties specified as 'required' should be defined"
     );
-    expect(res.warnings.length).toEqual(0);
```

Comment:
We need to have tests for all four scenarios: `allOf`, `oneOf`, `anyOf`, or none of the above.